containerd-shim-spin/src/engine.rs

```diff
@@ -182,57 +183,74 @@ impl SpinEngine {
         env::set_var("XDG_CACHE_HOME", &cache_dir);
         let app_source = self.app_source(ctx, &cache).await?;
         let resolved_app_source = self.resolve_app_source(app_source.clone(), &cache).await?;
-        let trigger_cmd = trigger_command_for_resolved_app_source(&resolved_app_source)
+        let trigger_cmds = trigger_command_for_resolved_app_source(&resolved_app_source)
             .with_context(|| format!("Couldn't find trigger executor for {app_source:?}"))?;
         let locked_app = self.load_resolved_app_source(resolved_app_source).await?;
-        self.run_trigger(&trigger_cmd, locked_app, app_source).await
+        self.run_trigger(
+            trigger_cmds.iter().map(|s| s.as_ref()).collect(),
+            locked_app,
+            app_source,
+        )
+        .await
     }
 
     async fn run_trigger(
```

Comment:
I wonder, the fact that we are blocking the futures in `run_wasi()` call, how would the concurrent performance be like for running multiple triggers at the same time.

Reply:
Hi @Mossaka ,

I ran some tests for this (using k3d cluster on my Macbook M2 pro):

(ran using `hey -z 10s http://localhost:8083/hello`)

| Scenario  | http-throughput  |
| ------------- | ------------- |
| v0.13.1 + only http-trigger app  | ~ 89k/s  |
| multi-trigger-shim + only http-trigger receiving traffic  | ~89k/s |
| multi-trigger-shim + both redis and http-trigger receiving traffic  | ~82k/s |

Reply:
okay that looks good. what's the throughput of the redis trigger in your last scenario?

Reply:
with both http & redis receiving traffic concurrently, I am able to inject about 1050 msgs (and I assume that also equates to number of events being processed by spin app)

with just redis receiving the traffic, the throughput is about 1450 msgs.